File: PHYTHON/Day7/credit_card_validator.py
def get_input(prompt_message):
	prompt = input('prompt_message')
	return prompt
def validate_user_input(input):
	if type(input) != str or any(value.isalpha() for value in input):
		raise TypeError("Must Pass In Only String of Numbers")
	return len(input) >= 13 and len(input) <= 16
def check_type_of_card(input):
	card_type = "Invalid Card"
	if input[0] == '4': card_type = "Visa Card"
	if input[0] == '5': card_type = "Master Card"
	if input[0] == '6': card_type = "Discover Card"
	if input[0] == '3' and input[1] == '7': card_type = "American Express Card"
	return card_type

def calculate_card_no(input):
	sum_of_odd_position = 0
	sum_of_even_position = 0
	for index,value in enumerate(input):
		element = (int) (value)
		square = 2 * element
		if square > 9:
			first_num = square % 10
			second_num = square // 10
			square = first_num + second_num
		if index % 2 == 0: sum_of_even_position += square
		else : sum_of_odd_position += element
	sum_card = sum_of_odd_position + sum_of_even_position
	return sum_card
		
def validate_card(input):
	if type(input) != int :
		raise TypeError('Only accpt Int')
	response = "Invalid"
	if(input % 10 == 0) : response = "Valid"
	return response
		

File: JAVASCRIPT/Day7/creditCardValidator.js
const prompt = require('prompt-sync')()

const getInput = (prompt_messsage) =>{
	let userInput = prompt(prompt_messsage)
	return  userInput
	}
const validateInput = (input) =>{
	check_length = input.length >= 13 && input.length <= 16
	return check_length

	}
const getCardNoToArrays = (input) =>{
	const array = input.split("");
	return array
	}

const getTypeOfCard = (input) =>{
	cardType = "Invalid Card"
	if(input.charAt(0) == "4") cardType = "Visa Card"
	if(input.charAt(0) == "5") cardType = "Master Card"
	if(input.charAt(0) == "6") cardType = "Discover Card"
	if(input.charAt(0) == "3" && input.charAt(1) == "7"  ) cardType = "American Express Card"
	return cardType 
	}

const calculateCard = (array)=>{
	sumOfOddPosition = 0
	sumOfEvenPosition = 0
	for (let count = array.length -1 ; count >= 0; count--){
		element = Number(array[count])
		getSquare = 2 * element
		if(getSquare > 9){
			getStringSquare = String(getSquare)
			getSquare = Number(getStringSquare.charAt(0)) + Number(getStringSquare.charAt(1))
			}
		if(count % 2 == 0) sumOfEvenPosition += getSquare 
		else sumOfOddPosition += element
			}
		sum = sumOfOddPosition + sumOfEvenPosition
		return sum;
	}

const validateCard = (sumOfCaluculateCard) =>{
	response = "Invalid"
	if(sumOfCaluculateCard % 10 == 0) response = "Valid"
	return response
	}

const mainInterface = () =>{
	console.log("***************************************************************")
	userInput =  getInput('Kindly Enter Your Credit Card No ')
	console.log(`Credit Card No : ${userInput}`)
	if(validateInput(userInput)){
	console.log(`Credit Card Digit Length : ${userInput.length}`)
	console.log(`Credit Card Type : ${getTypeOfCard(userInput )}`)
	getArrayCardNo = getCardNoToArrays(userInput)	
	calCard = calculateCard(getArrayCardNo)
	console.log(`Credit Card Validity Status :${validateCard(calCard )} `)		
			}
	else {	
	console.log(`Credit Card Digit Length :Must Be between 13 and 16`)
	console.log(`Credit Card Type : Invalid`)
	console.log(`Credit Card Validity Status :Invalid `)		

			}

	}
mainInterface() 